tests/TestCase/Validation/ValidatorTest.php

```diff
@@ -600,6 +600,34 @@ public function testCustomErrorsWithPresenceRequiredAsArray()
         $this->assertEquals($expected, $errors);
     }
 
+    /**
+     * Tests the testAllowEmptyFor method
+     *
+     * @return void
+     */
+    public function testAllowEmptyFor()
+    {
+        $validator = new Validator();
+        $validator
+            ->allowEmptyFor('title')
+            ->minLength('title', 5, 'Min. length 5 chars');
+
+        $results = $validator->errors(['title' => null]);
+        $this->assertSame([], $results);
+
+        $results = $validator->errors(['title' => '']);
+        $this->assertSame(['title' => ['minLength' => 'Min. length 5 chars']], $results);
+
+        $validator
+            ->allowEmptyFor('name', Validator::EMPTY_STRING);
+
+        $results = $validator->errors(['name' => null]);
+        $this->assertSame([], $results);
+
+        $results = $validator->errors(['name' => '']);
```

Comment:
Should there be a test for 0 and empty array as well?

Reply:
Done